kube-derive/src/custom_resource.rs

```diff
@@ -192,6 +198,122 @@ impl FromMeta for SchemaMode {
     }
 }
 
+/// This struct mirrors the fields of `k8s_openapi::CustomResourceSubresourceScale` to support
+/// parsing from the `#[kube]` attribute.
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Scale {
+    pub(crate) label_selector_path: Option<String>,
+    pub(crate) spec_replicas_path: String,
+    pub(crate) status_replicas_path: String,
+}
+
+// This custom FromMeta implementation is needed for two reasons:
+//
+// - To enable backwards-compatibility. Up to version 0.97.0 it was only possible to set scale
+//   subresource values as a JSON string.
+// - To be able to declare the scale sub-resource as a list of typed fields. The from_list impl uses
+//   the derived implementation as inspiration.
+impl FromMeta for Scale {
+    /// This is implemented for backwards-compatibility. It allows that the scale subresource can
+    /// be deserialized from a JSON string.
+    fn from_string(value: &str) -> darling::Result<Self> {
+        serde_json::from_str(value).map_err(darling::Error::custom)
+    }
+
+    fn from_list(items: &[darling::ast::NestedMeta]) -> darling::Result<Self> {
+        let mut errors = darling::Error::accumulator();
+
+        let mut label_selector_path: (bool, Option<Option<String>>) = (false, None);
+        let mut spec_replicas_path: (bool, Option<String>) = (false, None);
+        let mut status_replicas_path: (bool, Option<String>) = (false, None);
+
+        for item in items {
+            match item {
+                darling::ast::NestedMeta::Meta(meta) => {
+                    let name = darling::util::path_to_string(meta.path());
+
+                    match name.as_str() {
+                        "label_selector_path" => {
+                            if !label_selector_path.0 {
+                                let path = errors.handle(darling::FromMeta::from_meta(meta));
+                                label_selector_path = (true, Some(path))
+                            } else {
+                                errors.push(
+                                    darling::Error::duplicate_field("label_selector_path").with_span(&meta),
```

Comment:
from cross-referencing with k8s-openapi, this field is optional, but we make it mandatory with this
https://docs.rs/k8s-openapi/latest/k8s_openapi/apiextensions_apiserver/pkg/apis/apiextensions/v1/struct.CustomResourceSubresourceScale.html#structfield.label_selector_path

Reply:
This field is optional. The validation if a field is mandatory comes at a later stage in this function. This error is only returned if the field is set more than once.

Reply:
Ah. Ok! 🤦